examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala

```diff
@@ -0,0 +1,176 @@
+      */
+    private val CACHE_NAME = "testCache"
+
+    //Starting Ignite server node.
+    val ignite = setupServerAndData
+
+    closeAfter(ignite) { ignite ⇒
+        //Creating spark session.
+        implicit val spark = SparkSession.builder()
+            .appName("IgniteDataFrameJoinExample")
+            .master("local")
+            .config("spark.executor.instances", "2")
+            .getOrCreate()
+
+        // Adjust the logger to exclude the logs of no interest.
+        Logger.getRootLogger.setLevel(Level.ERROR)
+        Logger.getLogger("org.apache.ignite").setLevel(Level.INFO)
+
+        // Executing examples.
+
+        sparkDSLJoinExample
+        nativeSparkSqlJoinExample
+    }
+
+    /**
+      * Examples of usage Ignite DataFrame implementation.
+      * Selecting data throw Spark DSL.
+      *
+      * @param spark SparkSession.
+      */
+    def sparkDSLJoinExample(implicit spark: SparkSession): Unit = {
+        println("Querying using Spark DSL.")
+        println
+
+        val persons = spark.read
+          .format(FORMAT_IGNITE)
+          .option(OPTION_TABLE, "person")
+          .option(OPTION_CONFIG_FILE, CONFIG)
+          .load()
+
+        persons.printSchema()
+        persons.show()
+
+        val cities = spark.read
+          .format(FORMAT_IGNITE)
+          .option(OPTION_TABLE, "city")
+          .option(OPTION_CONFIG_FILE, CONFIG)
+          .load()
+
+        persons.printSchema()
+        persons.show()
+
+        val joinResult = persons.join(cities, persons("city_id") === cities("id"))
+          .select(persons("name").as("person"), persons("age"), cities("name").as("city"), cities("country"))
+
+        joinResult.explain(true)
+        joinResult.printSchema()
+        joinResult.show()
+    }
+
+    /**
+      * Examples of usage Ignite DataFrame implementation.
+      * Registration of Ignite DataFrame for following usage.
+      * Selecting data by Spark SQL query.
+      *
+      * @param spark SparkSession.
+      */
+    def nativeSparkSqlJoinExample(implicit spark: SparkSession): Unit = {
+        println("Querying using Spark SQL.")
+        println
+
+        val persons = spark.read
+          .format(FORMAT_IGNITE)
+          .option(OPTION_TABLE, "person")
+          .option(OPTION_CONFIG_FILE, CONFIG)
+          .load()
+
+        persons.printSchema()
+        persons.show()
+
+        val cities = spark.read
+          .format(FORMAT_IGNITE)
+          .option(OPTION_TABLE, "city")
+          .option(OPTION_CONFIG_FILE, CONFIG)
+          .load()
+
+        persons.printSchema()
+        persons.show()
+
+        // Registering DataFrame as Spark view.
+        persons.createOrReplaceTempView("person")
+        cities.createOrReplaceTempView("city")
+
+        // Selecting data from Ignite throw Spark SQL Engine.
+        val joinResult = spark.sql("SELECT person.name AS person, age, city.name AS city, country FROM person JOIN city ON person.city_id = city.id")
```

Comment:
Can we use multiline string literal here to make SQL query more readabale?

```
val joinResult = spark.sql(""" 
    | SELECT 
    |   person.name AS person,
    |   age,
    |   city.name AS city,
    |   country
    | FROM
    |   person JOIN 
    |   city ON person.city_id = city.id
""");
```

Reply:
Cool advise, thanks